signer/src/eth.rs

```diff
@@ -0,0 +1,346 @@
+    pub fn public_key(&self) -> ecdsa::PublicKey {
+        self.0.public_key()
+    }
+
+    /// Obtains the public address of the account by taking the last 20 bytes
+    /// of the Keccak-256 hash of the public key.
+    pub fn account_id(&self) -> AccountId20 {
+        let uncompressed = self.0 .0.public_key().serialize_uncompressed();
+        let hash = keccak(&uncompressed[1..]).0;
+        let hash20 = hash[12..].try_into().expect("should be 20 bytes");
+        AccountId20(hash20)
+    }
+
+    /// Signs an arbitrary message payload.
+    pub fn sign(&self, signer_payload: &[u8]) -> Signature {
+        let message_hash = keccak(signer_payload);
+        let wrapped =
+            Message::from_digest_slice(message_hash.as_bytes()).expect("Message is 32 bytes; qed");
+        Signature(crate::ecdsa::sign(&self.0 .0.secret_key(), &wrapped))
+    }
+}
+
+/// A signature generated by [`Keypair::sign()`].
+#[derive(Debug, Clone, Copy, PartialEq, Eq, codec::Encode)]
+pub struct Signature(pub [u8; 65]);
+
+impl AsRef<[u8; 65]> for Signature {
+    fn as_ref(&self) -> &[u8; 65] {
+        &self.0
+    }
+}
+
+/// A 20-byte cryptographic identifier.
+#[derive(Debug, Copy, Clone, PartialEq, Eq, codec::Encode)]
+pub struct AccountId20(pub [u8; 20]);
+
+impl AsRef<[u8]> for AccountId20 {
+    fn as_ref(&self) -> &[u8] {
+        &self.0
+    }
+}
+
+/// Verify that some signature for a message was created by the owner of the [`ecdsa::PublicKey`].
+///
+/// ```rust
+/// use subxt_signer::eth;
+///
+/// let keypair = eth::dev::alice();
+/// let message = b"Hello!";
+///
+/// let signature = keypair.sign(message);
+/// let public_key = keypair.public_key();
+/// assert!(eth::verify(&signature, message, &public_key));
+/// ```
+pub fn verify<M: AsRef<[u8]>>(sig: &Signature, message: M, pub_key: &ecdsa::PublicKey) -> bool {
+    let Ok(signature) = secp256k1::ecdsa::Signature::from_compact(&sig.0[..64]) else {
+        return false;
+    };
+    let message_hash = keccak(message.as_ref());
+    let wrapped =
+        Message::from_digest_slice(message_hash.as_bytes()).expect("Message is 32 bytes; qed");
+    let pub_key = secp256k1::PublicKey::from_slice(&pub_key.0).expect("valid public key");
+
+    Secp256k1::verification_only()
+        .verify_ecdsa(&wrapped, &signature, &pub_key)
+        .is_ok()
+}
+
+/// An error handed back if creating the keypair fails.
+#[derive(Debug, PartialEq, Display, From)]
+pub enum Error {
+    /// Invalid private key.
+    #[display(fmt = "Invalid private key")]
+    #[from(ignore)]
+    InvalidPrivateKey,
+    /// Invalid hex.
+    #[display(fmt = "Cannot parse hex string: {_0}")]
+    Hex(hex::FromHexError),
+    /// Inner,
+    #[display(fmt = "{_0}")]
+    Inner(ecdsa::Error),
+}
+
+#[cfg(feature = "std")]
+impl std::error::Error for Error {}
+
+/// Dev accounts, helpful for testing but not to be used in production,
+/// since the secret keys are known.
+pub mod dev {
```

Comment:
Can you confirm whether moonbeam uses these dev keys at all? Mythos doesn't; they have their own well known private keys for dev accounts.

If moonbeam uses these dev accounts then let's keep them, else if the well known keys are shared between eth impls let's use those instead, else let's ditch them.

The readme gives account details but also says that they come from substrate keys, so we can generate the test accounts and check that they match here if not done already: https://github.com/moonbeam-foundation/moonbeam/blob/e70ee0d427dfee8987d5a5671a66416ee6ec38aa/README.md?plain=1#L56

Reply:
By use you mean its prefunded on their devnet?